QQE-379.md

```diff
@@ -0,0 +1,44 @@
+# QQE-379 - HTTP module / REST use-cases - Development topics - JSON payload
+## Links
+
+Jira : [QQE-379](https://issues.redhat.com/browse/QQE-379)
+TD:
+
+This test plan outlines the testing for different JSON payloads in Quarkus REST endpoints.
+It aims to ensure the correct handling of JSON data in various scenarios, including successful processing, error handling, and complex data structures.
+
+## Scope of the testing
+The test coverage will be verified for JVM, native mode and Openshift.
+
+### What to test
+
+ - Successful JSON payload processing: Verify that valid JSON payloads requests to REST endpoints and assert that the responses contain the correct JSON data and status codes.
+ - Check Big JSON payload processing: Verify that the application can efficiently process large JSON payloads without issues.
+ - Handling invalid JSON payloads: Test how the application responds to malformed or invalid JSON requests,
+ensuring appropriate error responses are returned.
+ - Handling missing or incorrect data types: Validate the behavior when required JSON fields are missing or have incorrect data types,
+ensuring proper error handling.
+ - Testing complex JSON structures: Evaluate the handling of JSON payloads containing nested objects, arrays, and various data types,
+ensuring serialization and deserialization.
+
+### What not to test
+
+- Specific JSON Library Configurations: This test plan focuses on the functionality of JSON handling in Quarkus REST endpoints and clients,
+not the specifics of underlying JSON libraries (e.g., Jackson, JSON-B). While both libraries will be used during testing, the aim is to verify the behavior of the Quarkus.
```

Comment:
Are both of them supported? If they're both supported and recommended, imo we should take them as valid part of testing 'matrix'

If Quarkus is heavily opinionated here or doesn't allow for configuration, I think my point is moot but if they're both supported and configurable, we should be testing it

Reply:
Yes, currently both are supported, although seems from docs that they use Jackson as a first option. And also, I checked this "old" thread discussion--> https://groups.google.com/g/quarkus-dev/c/RTnvH6LSdxs   where they lean towards Jackson. 

Anyways, I could include both as you mentioned.

Reply:
Was there any conclusion on this? I'm gonna unresolve, but maybe I am misunderstanding something.

Reply:
Well, I was investigating a bit, and my idea was introduce the TD into the http/http-advance-reactive module and there already is used `quarkus-rest-jsonb` so it's used for serialization support for Quarkus REST.
On the other hand, Jackson serialization quarkus-rest-jackson, it's already used in other tests as in http/jakarta-rest-reactive with advanced features such as `JsonView support` or `@CustomSerialization`.
So from my humble opinion, I would put out of this testing scope .
However, I am open to incorporating alternative approaches or additional tests if you have any suggestions about it.

Reply:
I was thinking also, if I could execute same tests using jackson library, we will have the coverage for both libraries, but not sure if I can get it exactly.